docs/aws/audit/elbmonitoring/rules/min_ec2_instances_available.mdx

```diff
@@ -24,6 +24,261 @@ CBP
+   ```sh
+   aws elb describe-load-balancers
+   ```
+
+2. **Describe Instances for Each ELB:**
+   For each ELB, list the instances currently registered with it. Replace `<load-balancer-name>` with the name of your ELB.
+   ```sh
+   aws elb describe-instance-health --load-balancer-name <load-balancer-name>
+   ```
+
+3. **Register Instances with ELB:**
+   If the number of instances is below the desired minimum, register additional instances with the ELB. Replace `<load-balancer-name>` with the name of your ELB and `<instance-id>` with the ID of the instance you want to add.
+   ```sh
+   aws elb register-instances-with-load-balancer --load-balancer-name <load-balancer-name> --instances <instance-id>
+   ```
+
+4. **Automate Instance Registration:**
+   To ensure that the minimum number of instances is always maintained, consider using an Auto Scaling Group (ASG) with your ELB. This will automatically manage the number of instances. Create or update an ASG and attach it to your ELB. Replace `<auto-scaling-group-name>`, `<launch-configuration-name>`, and `<load-balancer-name>` with your specific names.
+   ```sh
+   aws autoscaling create-auto-scaling-group --auto-scaling-group-name <auto-scaling-group-name> --launch-configuration-name <launch-configuration-name> --min-size <min-instances> --max-size <max-instances> --load-balancer-names <load-balancer-name>
+   ```
+
+By following these steps, you can ensure that your ELBs always have the minimum required number of EC2 instances registered, thereby maintaining high availability and performance.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of having fewer than the minimum number of EC2 instances configured for Elastic Load Balancers (ELBs) in AWS using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK (Boto3) and Authentication:**
+   - Install the Boto3 library if you haven't already.
+   - Configure your AWS credentials.
+
+   ```bash
+   pip install boto3
+   ```
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon EC2
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+
+   # Initialize the ELB and EC2 clients
+   elb_client = session.client('elb')
+   ec2_client = session.client('ec2')
+   ```
+
+2. **Retrieve ELB and Associated Instances:**
+   - Fetch the list of all ELBs and their associated instances.
+
+   ```python
+   def get_elb_instances(elb_client):
+       elbs = elb_client.describe_load_balancers()
+       elb_instances = {}
+       for elb in elbs['LoadBalancerDescriptions']:
+           elb_name = elb['LoadBalancerName']
+           instances = elb['Instances']
+           elb_instances[elb_name] = [instance['InstanceId'] for instance in instances]
+       return elb_instances
+
+   elb_instances = get_elb_instances(elb_client)
+   ```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>